src/paulscode/android/mupen64plusae/input/map/VisibleTouchMap.java

```diff
@@ -551,12 +551,12 @@ private void loadFpsIndicator( Profile profile )
      */
     private void loadAutoHoldImages( Profile profile, String name )
     {
-        Matcher matcher = Pattern.compile( "([^-]*)-hold(.*)" ).matcher( name );
-        if( matcher.groupCount() < 3 )
+        if ( !name.contains("-hold") )
             return;
         
-        String group = matcher.group( 1 );
-        String hold = matcher.group( 2 );
+        String[] fields = name.split( "-hold" );
+        String group = fields[0];
+        String hold = fields[1];
         
```

Comment:
Ah, thanks for tracking this down.  The bug is that I should have tested `!matcher.matches()` rather than `matcher.groupCount(0) < 3`.

I forget why I used a regex in the first place.  Maybe a carryover from when the parsing was more complex (the touchscreen stuff changed a lot between v2 and v3).  Your fix is easier to read than a regex, so I'll merge this.

Thanks :)